docs/dev/graphics-stack.md

```diff
@@ -0,0 +1,398 @@
+# init 6
+```
+
+## gfx-drm
+
+The gfx-drm gate consists of different kernel components for support of various graphic adapters in the illumos and the Direct Rendering Infrastructure (DRI) library, libdrm.
+
+Component  | Description
+---------  | ------------
+libdrm     | The library enables communication between kernel components and user-space libraries through the DRI protocol.
+agpart     | Kernel driver for the Graphics Address Remapping Table (GART) / Graphics Translation Table (GTT) support.
+header-drm | System headers for DRM/KMS kernel drivers and user-space libraries.
+i915       | Intel KMS driver for Intel Graphics Media Accelerator (GMA) and Intel HD Graphics adapters.
+
+### Intel KMS driver
+
+The xorg-video-intel package is the open-source 2D graphics driver for the X Window System as implemented by X.org. It supports a variety of Intel graphics chipsets including:
+
+* i810/i810e/i810-dc100,i815,
+* i830M,845G,852GM,855GM,865G,
+* 915G/GM,945G/GM/GME,946GZ
+* G/GM/GME/Q965,
+* G/Q33,G/Q35,G41,G/Q43,G/GM/Q45
+* PineView-M (Atom N400 series)
+* PineView-D (Atom D400/D500 series)
+* Intel(R) HD Graphics,
+* Intel(R) Iris(TM) Graphics,
+* Intel(R) Iris(TM) Pro Graphics.
+
+Open source kernel driver(s) by Oracle:
+
+<https://github.com/oracle/solaris-xorg/tree/master/open-src/kernel>
+
+Relevant kernel code is located in drm and i915 subfolders.
+
+illumos port of Oracle code is available here:
+
+<https://github.com/illumos/gfx-drm>
+
+Martin Bochnig, creator of OpenSXCE, initially backported the S12 driver to illumos, available in oi-userland until December 20. 2016.
+
+It was then superseeded by the new gate providing improvements to the gfx_private interface and agpgart driver, as well as containing userland libraries used by graphics, like libdrm.
+
+#### References
+
+Alan Coopersmith pointed to the following documents regarding DRI/KMS:
+
+* <http://www.phoronix.com/scan.php?page=news_item&px=Solaris-DRM-KMS-2015>
+* <http://www.x.org/wiki/Events/XDC2015/Program/#Fishel_status_drm_i915_solaris>
+* <https://en.wikipedia.org/wiki/Direct_Rendering_Manager>
+* <http://dri.freedesktop.org/wiki/>
+* <https://wiki.archlinux.org/index.php/kernel_mode_setting>
+* <http://lanyrd.com/topics/x-window-system>
+
+Additionally, information about driver development:
+
+* <https://01.org/linuxgraphics>
+* <http://dri.freedesktop.org/docs/drm/>
+* <http://blog.ffwll.ch/2013/01/i915gem-crashcourse-overview.html>
+* blog posts from the i915 Linux maintainer <http://blog.ffwll.ch/>
+
+#### Known Issues
+
+##### SNA
+
+May causes segmentation fault if enabled. If so, just disable SNA and you can copy attached 20-intel-uxa.conf to /etc/X11/xorg.conf.d/.
+
+For now, UXA mode is set by default.
+
+#### Sandy Bridge GT1
+
+A hardware bug requires implementing a workaround in the Intel ringbuffer implementation, occasional 1-2 second hangs may occur.
+
+#### Gen >= 8
+
+Generation 8 specifics in Intel ringbuffer are not implemented yet.
```

Comment:
Better explicitly say what is hidden in the table below: "Gen >=8 (Broadwell and newer) not supported at all."